dll/win32/shell32/shlexec.cpp

```diff
@@ -2452,11 +2452,16 @@ HRESULT WINAPI ShellExecCmdLine(
         }
     }
 
-    if (PathIsURLW(lpCommand) || UrlIsW(lpCommand, URLIS_APPLIABLE))
+    dwSize = _countof(szFile);
+    if (UrlIsW(lpCommand, URLIS_URL))
     {
         StringCchCopyW(szFile, _countof(szFile), lpCommand);
         pchParams = NULL;
     }
+    else if (UrlApplySchemeW(lpCommand, szFile, &dwSize, URL_APPLY_GUESSSCHEME) == S_OK)
```

Comment:
Shouldn't an URL scheme be applied only if the string is an URL? 🤔

Reply:
`www.google.de` isn't a perfect URL but it will be interpreted as a URL.

Reply:
I mean, does it also pass the `UrlIsW()` test above? or not?

Reply:
Nope.

Reply:
![UrlIsTest](https://user-images.githubusercontent.com/2107452/122341102-10d2a000-cf7e-11eb-80f8-bda4550d9569.png)

Reply:
![image](https://user-images.githubusercontent.com/2107452/122342848-06b1a100-cf80-11eb-834a-100bb77206fa.png)
`ShellExecuteEx` opens the browser for `www.google.de`. Do we have to do it in `ShellExecuteEx`?